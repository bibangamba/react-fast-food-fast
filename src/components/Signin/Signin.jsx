import React from "react";
import PropTypes from "prop-types";
import { MDBContainer, MDBRow, MDBCol, MDBBtn } from "mdbreact";

const Signin = ({
  onChange,
  onSubmit,
  signinButtonText,
  signinButtonDisabled
}) => (
  <MDBContainer className="mt-5">
    <MDBRow className="justify-content-center">
      <MDBCol md="6" className="mt-4">
        <form onSubmit={onSubmit} className="w-100 ">
          <p className=" text-black-50 h4 text-center mb-1">Sign in</p>
          <label
            htmlFor="defaultFormSigninEmailEx"
            className="text-black-50 w-100"
          >
            Email address
            <input
              type="email"
              name="email"
              onChange={onChange}
              id="defaultFormSigninEmailEx"
              className="form-control"
            />
          </label>
          <br />
          <label
            htmlFor="defaultFormSigninPasswordEx"
            className="text-black-50 w-100"
          >
            Password
            <input
              name="password"
              type="password"
              onChange={onChange}
              id="defaultFormSigninPasswordEx"
              className="form-control"
            />
          </label>

          <div className="text-center mt-4">
            <MDBBtn
              disabled={signinButtonDisabled}
              color="rgba-blue-strong"
              className="rgba-blue-strong"
              type="submit"
            >
              {signinButtonText}
            </MDBBtn>
          </div>
        </form>
      </MDBCol>
    </MDBRow>
  </MDBContainer>
);

Signin.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  signinButtonDisabled: PropTypes.bool,
  signinButtonText: PropTypes.string
};

Signin.defaultProps = {
  signinButtonDisabled: false,
  signinButtonText: "Login"
};

export default Signin;
