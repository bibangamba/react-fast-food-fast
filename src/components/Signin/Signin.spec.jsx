import { shallow } from "enzyme";
import React from "react";
import Signin from "./Signin";

const props = {
  onChange: jest.fn(),
  onSubmit: jest.fn(),
  error: ""
};
describe("Signin test", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Signin {...props} />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
