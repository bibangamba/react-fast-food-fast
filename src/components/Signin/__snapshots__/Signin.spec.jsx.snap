// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin test matches snapshot 1`] = `ShallowWrapper {}`;
